### PR TITLE
Feature motor encoder pololu 12 v

### PR DESCRIPTION
**Pull request by Joepieler**
The Motor encoder has been added to Turn the Beetle.

### CODE REVIEW

File: code/src/beetle.cpp [hunk 1]
```diff
@@ -28,47 +31,86 @@ namespace r2d2::moving_platform {
...
     }
     void beetle_c::turn(int16_t degrees) {
-        int new_degrees = degrees;
         // because of inaccuracies, when moving forward, the steer of manual
         // control will give a number beween -10 and 10 Just to make sure the
         // robot will move forward and not react to quickly, a theshold is made
         // for -15 till 15.
-        if (new_degrees < 15 && new_degrees > -15) {
-            new_degrees = 0;
+        if (degrees >= 15 && degrees <= 90) {
```
Comment (TGAssen) [inline code/src/beetle.cpp]: Boundary values should be variables

File: code/src/beetle.cpp [hunk 2]
```diff
@@ -28,47 +31,86 @@ namespace r2d2::moving_platform {
...
+        // Encode frequency for 1 turn of the wheel. the encoder has 64 point
+        // per over 2 pins we count when de adc goes from low to high of 1 pin.
+        // The gear ratio from the motor is 50:1 64/4*50 = 800
+        int encode_1_full_turn = 800;
+        // Motor speed
+        int motor_speed;
+        // Turn the right way.
+        if (degrees > 0) {
+            // turn right
+            motor_speed = 20;
```
Comment (TGAssen) [inline code/src/beetle.cpp]: 20 is a magic value

File: code/src/beetle.cpp [hunk 3]
```diff
@@ -28,47 +31,86 @@ namespace r2d2::moving_platform {
...
     void beetle_c::turn(int16_t degrees) {
-        int new_degrees = degrees;
         // because of inaccuracies, when moving forward, the steer of manual
         // control will give a number beween -10 and 10 Just to make sure the
         // robot will move forward and not react to quickly, a theshold is made
         // for -15 till 15.
-        if (new_degrees < 15 && new_degrees > -15) {
-            new_degrees = 0;
+        if (degrees >= 15 && degrees <= 90) {
+            degrees = degrees;
```
Comment (TGAssen) [inline code/src/beetle.cpp]: does anything even change here?

Comment (Joepieler) [reply code/src/beetle.cpp]: variable name, we don't use new_degrees anymore

File: code/src/beetle.cpp [hunk 4]
```diff
@@ -28,47 +30,82 @@ namespace r2d2::moving_platform {
...
+        int counter_m0 = 0;
+        bool low_m1 = false;
+        int counter_m1 = 0;
+        // The adc input. is between 3000 and 3800.
+        unsigned int adc_voltage = 3500;
+        // Encode frequency for 1 turn of the wheel. the encoder has 64 point
+        // per over 2 pins we count when de adc goes from low to high of 1 pin.
+        // The gear ratio from the motor is 50:1 64/4*50 = 800
+        int encode_1_full_turn = 800;
+        // Motor speed
```
Comment (TGAssen) [inline code/src/beetle.cpp:59]: Comment unclear, What are the boundary values of motorspeed? if there are any at all, or is it a relative value?

File: code/src/beetle.cpp [hunk 5]
```diff
@@ -4,9 +4,11 @@ namespace r2d2::moving_platform {
...
-        base_comm_c &comm)
+        base_comm_c &comm, hwlib::adc &encode_m0, hwlib::adc &encode_m1)
         : moving_platform_c(comm),
-          qik_2s12v10_motorcontroller(qik_2s12v10_motorcontroller) {
+          qik_2s12v10_motorcontroller(qik_2s12v10_motorcontroller),
+          encode_m0(encode_m0),
+          encode_m1(encode_m1) {
     }
     void beetle_c::set_speed(int8_t new_speed) {
         // Todo: depending on the motor configuration, 127 may not be the
```
Comment (TGAssen) [inline code/src/beetle.cpp]: should a todo be present in a pull request?

**Review by TGAssen — APPROVED** [review]
The code complies with the guide lines setup for the r2d2 project.

**Pull request merged**